pkg/execute/kubectl/merger.go

```diff
@@ -0,0 +1,138 @@
+package kubectl
+
+import (
+	"github.com/kubeshop/botkube/pkg/config"
+)
+
+// EnabledKubectl configuration for executing commands inside cluster
+type EnabledKubectl struct {
+	AllowedKubectlVerb     map[string]struct{}
+	AllowedKubectlResource map[string]struct{}
+
+	DefaultNamespace string
+	RestrictAccess   bool
+}
+
+// Merger provides functionality to merge multiple bindings
+// associated with the kubectl executor.
+type Merger struct {
+	executors config.IndexableMap[config.Executors]
+}
+
+// NewMerger returns a new Merger instance.
+func NewMerger(executors config.IndexableMap[config.Executors]) *Merger {
+	return &Merger{
+		executors: executors,
+	}
+}
+
+// MergeForNamespace returns kubectl configuration for a given set of bindings.
+//
+// It merges entries only if a given Namespace is matched.
+//   - kubectl.commands.verbs     - strategy append
+//   - kubectl.commands.resources - strategy append
+//   - kubectl.defaultNamespace   - strategy override (if not empty)
+//   - kubectl.restrictAccess     - strategy override (if not empty)
+//
+// The order of merging is the same as the order of items specified in the includeBindings list.
+func (kc *Merger) MergeForNamespace(includeBindings []string, forNamespace string) EnabledKubectl {
+	enabledInNs := func(executor config.Kubectl) bool {
+		return executor.Enabled && executor.Namespaces.IsAllowed(forNamespace)
+	}
+	return kc.merge(kc.collect(includeBindings, enabledInNs))
+}
+
+// MergeAllEnabled returns kubectl configuration for all kubectl configs.
+func (kc *Merger) MergeAllEnabled(includeBindings []string) EnabledKubectl {
+	onlyEnabled := func(executor config.Kubectl) bool {
+		return executor.Enabled
+	}
+	return kc.merge(kc.collect(includeBindings, onlyEnabled))
+}
+
+// IsAtLeastOneEnabled returns true if at least one kubectl executor is enabled.
+func (kc *Merger) IsAtLeastOneEnabled() bool {
+	for _, executor := range kc.executors {
+		if executor.Kubectl.Enabled {
+			return true
+		}
+	}
+	return false
+}
+
```

Comment:
Having had a cursory look at `MergeAllEnabledVerbs` is used.

To get the semantics right, what if it simply returns a Set? (`map[string]bool`) ?

As a freebie, you just get the value and wont't need `_, ok := ...`

Reply:
I learned that a long time ago. I'm using the `map[string]struct{}` not because it's give us [allocation improvements](https://dave.cheney.net/2014/03/25/the-empty-struct) but it gives more explicit API 



>I use and will continue to use map[string]struct{}, because when someone reads that, they know, beyond a shadow of a doubt, that existence is the only thing that matters and there is no place in the code, no matter how far this map travels, that will ever make a decision based on anything else. When I used map[string]bool, the person reading the code does NOT know that. You can assert it in a comment all you like, but the reader must still consult the code to verify that it can't go anywhere where some other programmer (who may even be you at some other point in time) didn't decide to make the bool parameter mean something. 

So when I see `map[string]bool` I start thinking if there is a functionality that is able to enable/disable it, if yes then on what condition? 
On the other hand, using `[]string` give us O(n) instead of `O(1)`. 

I'm open to discuss it further and establish an approach that would be suitable for both of us 👍

Reply:
All good. 👍